src/typecheck/arrays.ml

```diff
@@ -1,6 +1,42 @@
 
 module Id = Dolmen.Std.Id
 
+(* Ae arrays *)
+(* ************************************************************************ *)
+
+module Ae = struct
+
+  module Tff
+      (Type : Tff_intf.S)
+      (Ty : Dolmen.Intf.Ty.Ae_Array with type t := Type.Ty.t)
+      (T : Dolmen.Intf.Term.Ae_Array with type t := Type.T.t) = struct
+
+    type _ Type.err +=
+      | Bad_farray_arity : Dolmen.Std.Term.t Type.err
+
+    let parse env s =
+      match s with
+      | Type.Id { name = Simple "farray"; ns = Term } ->
+        `Ty (
+          fun ast args ->
+            match args with
+            | [ty] ->
+              let int_ty = Dolmen_std.Term.builtin Dolmen_std.Term.Int () in
+              Ty.array (Type.parse_ty env int_ty) (Type.parse_ty env ty)
```

Comment:
You shouldn't need to go through the type-checker to type the `int` constant. Instead, you should require the int type in the `Ty` functor argument.

Reply:
Done.